src/coreComponents/constitutive/solid/porosity/PorosityBase.hpp

```diff
@@ -0,0 +1,205 @@
+   */
+  GEOSX_HOST_DEVICE
+  localIndex numGauss() const { return m_newPorosity.size( 1 ); }
+
+  PorosityBaseUpdates( arrayView2d< real64 > const & newPorosity,
+                       arrayView2d< real64 > const & oldPorosity,
+                       arrayView2d< real64 > const & dPorosity_dPressure,
+                       arrayView1d< real64 > const & referencePorosity ):
+    m_newPorosity( newPorosity ),
+    m_oldPorosity( oldPorosity ),
+    m_dPorosity_dPressure( dPorosity_dPressure ),
+    m_referencePorosity ( referencePorosity )
+  {}
+
+  /// Default copy constructor
+  PorosityBaseUpdates( PorosityBaseUpdates const & ) = default;
+
+  /// Default move constructor
+  PorosityBaseUpdates( PorosityBaseUpdates && ) = default;
+
+  /// Deleted copy assignment operator
+  PorosityBaseUpdates & operator=( PorosityBaseUpdates const & ) = delete;
+
+  /// Deleted move assignment operator
+  PorosityBaseUpdates & operator=( PorosityBaseUpdates && ) = delete;
+
+  /**
+   * @brief Helper to save point stress back to m_newPorosity array
+   *
+   * This is mostly defined for improving code readability.
+   *
+   * @param[in] k Element index.
+   * @param[in] q Quadrature point index.
+   * @param[in] porosity porosity to be saved to m_newPorosity[k][q]
+   */
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  void savePorosity( localIndex const k,
+                     localIndex const q,
+                     real64 const & porosity,
+                     real64 const & dPorosity_dPressure ) const
+  {
+    m_newPorosity[k][q] = porosity;
+    m_dPorosity_dPressure[k][q] = dPorosity_dPressure;
+  }
+
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  real64 getPorosity( localIndex const k,
+                      localIndex const q ) const
+  {
+    return m_newPorosity[k][q];
+  }
+
+
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  real64 getOldPorosity( localIndex const k,
+                         localIndex const q ) const
+  {
+    return m_oldPorosity[k][q];
+  }
+
+protected:
+  arrayView2d< real64 > m_newPorosity;
+
+  arrayView2d< real64 > m_oldPorosity;
+
+  arrayView2d< real64 > m_dPorosity_dPressure;
+
+  arrayView1d< real64 > m_referencePorosity;
+};
+
+
+class PorosityBase : public ConstitutiveBase
+{
+public:
+  PorosityBase( string const & name, Group * const parent );
+
+  virtual ~PorosityBase() override;
+
+  std::unique_ptr< ConstitutiveBase > deliverClone( string const & name,
+                                                    Group * const parent ) const override;
+
+  virtual void allocateConstitutiveData( dataRepository::Group & parent,
+                                         localIndex const numConstitutivePointsPerParentIndex ) override;
+
+  static string catalogName() { return "PorosityBase"; }
+
+  virtual string getCatalogName() const override { return catalogName(); }
+
+  struct viewKeyStruct : public ConstitutiveBase::viewKeyStruct
+  {
+    static constexpr char const * newPorosityString() { return "porosity"; }
```

Comment:
This should not not be `newPorosityString` but `porosityString` to stay consistent.

Reply:
I followed what was done for the stress but I am happy to change it.